src/NuGet.Clients/NuGet.CommandLine/Commands/ClearCacheCommand.cs

```diff
@@ -0,0 +1,140 @@
+            var localAppDataFolderPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            var httpCacheFolderPath = Path.Combine(localAppDataFolderPath, "NuGet", "v3-cache");
+            if (!string.IsNullOrEmpty(httpCacheFolderPath))
+            {
+                Console.WriteLine(
+                    LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_ClearingNuGetHttpCache)),
+                    httpCacheFolderPath);
+
+                var result = ClearCacheDirectory(httpCacheFolderPath);
+                if (commandResult == 0)
+                {
+                    commandResult = result;
+                }
+            }
+
+            if (ClearGlobalPackages)
+            {
+                // also clear the global NuGet packages cache
+                var globalPackagesFolderPath = SettingsUtility.GetGlobalPackagesFolder(Settings);
+
+                Console.WriteLine(
+                    LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_ClearingNuGetGlobalPackagesCache)),
+                    globalPackagesFolderPath);
+
+                var result = ClearCacheDirectory(globalPackagesFolderPath);
+                if (commandResult == 0)
+                {
+                    commandResult = result;
+                }
+            }
+
+            if (commandResult != 0)
+            {
+                throw new CommandLineException(LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_CacheClearFailed)));
+            }
+
+            return Task.FromResult(commandResult);
+        }
+
+        private int ClearCacheDirectory(string folderPath)
+        {
+            // Calling DeleteRecursive rather than Directory.Delete(..., recursive: true)
+            // due to an infrequent exception which can be thrown from that API
+            var failedDeletes = new List<string>();
+            DeleteRecursive(folderPath, failedDeletes);
+
+            if (failedDeletes.Any())
+            {
+                Console.WriteWarning(
+                    LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_CachePartiallyCleared)));
+
+                foreach (var failedDelete in failedDeletes.OrderBy(f => f, StringComparer.OrdinalIgnoreCase))
+                {
+                    Console.WriteWarning(
+                        LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_FailedToDeletePath)),
+                        failedDelete);
+                }
+                return 1;
+            }
+            else
+            {
+                Console.WriteLine(LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_CacheCleared)));
+                return 0;
+            }
+        }
+
+        private static void DeleteRecursive(string deletePath, List<string> failedDeletes)
+        {
+            if (!Directory.Exists(deletePath))
```

Comment:
No, because you still use recursion into the folder structure. If you want to change the readonly first, you can enumerate all the files recursively and remove the readonly mark first. Then end up with non recursive code, which is a lot cleaner.

Your code will still go into symbolic links as it stands right now See line 120